ovirt/resource_ovirt_vm.go

```diff
@@ -1609,3 +1631,48 @@ func getTemplateDiskAttachments(templateID string, meta interface{}) ([]*ovirtsd
 	}
 	return nil, nil
 }
+
+func getAffinityGroups(conn *ovirtsdk4.Connection, cID string, agNames []interface{}) (ag []*ovirtsdk4.AffinityGroup, err error) {
+	var ags []*ovirtsdk4.AffinityGroup
+	var notFoundAGs []string
+
+	res, err := conn.SystemService().ClustersService().
+		ClusterService(cID).AffinityGroupsService().
+		List().Send()
+	if err != nil {
+		return nil, err
+	}
+	agNamesMap := make(map[string]*ovirtsdk4.AffinityGroup)
+	for _, af := range res.MustGroups().Slice() {
+		agNamesMap[af.MustName()] = af
+	}
+	for _, agName := range agNames {
+		if _, ok := agNamesMap[agName.(string)]; !ok {
+			notFoundAGs = append(notFoundAGs, agName.(string))
+		} else {
+			ags = append(ags, agNamesMap[agName.(string)])
+		}
+	}
+	if len(notFoundAGs) > 0 {
+		return nil, fmt.Errorf("affinity groups %v was not found on cluster %v", notFoundAGs, cID)
+	}
+	return ags, nil
+}
+
+func addVmToAffinityGroups(conn *ovirtsdk4.Connection, vm *ovirtsdk4.Vm, cID string, ags []*ovirtsdk4.AffinityGroup) error {
+	for _, ag := range ags {
+		log.Printf("Adding machine %v to affinity group %v", vm.MustName(), ag.MustName())
```

Comment:
Both parameters seem to be strings, they should use the formatting characters `%s`.